qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -923,7 +927,7 @@ def _copy_metadata(
             and metadata but different ID.
         """
         if new_instance is None:
-            new_instance = self.__class__()
+            new_instance = DbExperimentDataV1()
```

Comment:
This comment made me look at `_copy_metadata`, and I realize that we have to override it in `CompositeExperimentData`. I'm working on it now. It's delicate, because we store information with experiment ids in several places, and they're not supposed to be copied: in tags (parent and root ids), and in metadata (component ids and classes).